### PR TITLE
Improved types in tests (preparing for PHPStan level 6)

### PR DESCRIPTION
**Pull request by aivchen**
On phpstan level 6 there are around 700 errors in the project. I think it will be better to fix them in batches. What do you think?

### CODE REVIEW

**Comment by dantleech** [comment]
Hmm, should probably merge #1047 before going too far down this path

**Comment by aivchen** [comment]
Oh, ok. Let's wait:)

File: lib/Registry/Config.php [hunk 1]
```diff
@@ -29,8 +29,13 @@ class Config extends ArrayObject
      */
     final public const NAME_REGEX = '{^[0-9a-zA-Z_-]+$}';
 
+    /** @var string */
     private $name;
 
+    /**
+     * @param string $name
+     * @param mixed[] $config
```
Comment (dantleech) [inline lib/Registry/Config.php:37]: not important but i guess this is the same as `array<array-key,mixed>`, but it really should always be `array<string,mixed>` IIRC

File: lib/Registry/ConfigurableRegistry.php [hunk 2]
```diff
@@ -63,7 +63,7 @@ public function __construct(
     /**
      * Return the named configuration.
      *
-     * @param string|array $name
+     * @param string|mixed[] $name
```
Comment (dantleech) [inline lib/Registry/ConfigurableRegistry.php:66]: should be `string[]`

Comment (dantleech) [inline lib/Registry/ConfigurableRegistry.php:66]: should be `string|array<string,mixed>` (i.e. the `config` record)

File: lib/Registry/ConfigurableRegistry.php [hunk 3]
```diff
@@ -113,6 +113,8 @@ public function getConfigNames(): array
      * Note that all configurations must be associated with a named service
      * via a configuration key equal to the configuration service type of this registry.
      *
+     * @param mixed[] $config
```
Comment (dantleech) [inline lib/Registry/ConfigurableRegistry.php:116]: array<string,mixed>

File: tests/Benchmark/Console/Command/RunCommandBench.php [hunk 4]
```diff
@@ -63,6 +63,10 @@ class NothingBench { public function benchNothing(): void {}}
         );
     }
 
+    /**
+     * @param mixed[] $args
+     * @param mixed[] $config
```
Comment (dantleech) [inline tests/Benchmark/Console/Command/RunCommandBench.php:68]: $args => array<string,mixed>` config also

File: tests/Benchmark/IntegrationBenchCase.php [hunk 5]
```diff
@@ -19,6 +19,9 @@ protected function workspace(): Workspace
         return Workspace::create(__DIR__ . '/../Workspace');
     }
 
+    /**
+     * @param mixed[] $config
+     */
     protected function container(array $config = []): ContainerInterface
```
Comment (dantleech) [inline tests/Benchmark/IntegrationBenchCase.php:25]: ditto

**Comment by dantleech** [comment]
looks good. almost all the `$config` things should be `array<string,mixed>` (i.e. a "record) and a few other things are too broadly typed, but not sure it would pay too be too strict at this point in time. if we can make phpstan happy at level 6 it's already good :slightly_frowning_face:

**Comment by aivchen** [comment]
> looks good. almost all the `$config` things should be `array<string,mixed>`

You didn't give me a chance to fix it:))

>if we can make phpstan happy at level 6 it's already good

Yup, at level 6 we still can make some tricks. But at higher levels we will have to get rid of the mixed type.

**Pull request merged**